storage/src/node/branch.rs

```diff
@@ -50,45 +54,34 @@ impl BranchNode {
     /// The maximum number of children in a [BranchNode]
     pub const MAX_CHILDREN: usize = 16;
 
-    /// Obtain a mutable reference to a child address within a branch
-    /// This convenience method takes a u8 as the nibble offset
-    /// Panics if `child_index` >= [BranchNode::MAX_CHILDREN].
-    pub fn child_mut(&mut self, child_index: u8) -> &mut Option<LinearAddress> {
-        self.children
-            .get_mut(child_index as usize)
-            .expect("child_index is in bounds")
-    }
-
-    /// Returns the address of the child at the given index.
+    /// Returns the address of the child at the given index, and the hash,
+    /// which is None if we don't know the hash.
     /// None if there is no child at that index.
     /// Panics if `child_index` >= [BranchNode::MAX_CHILDREN].
-    pub fn child(&self, child_index: u8) -> Option<&LinearAddress> {
+    pub fn child(&self, child_index: u8) -> Option<(&LinearAddress, Option<&TrieHash>)> {
```

Comment:
Would it be better to just return `Option<(LinearAddress, Option<TrieHash>)>`? 

`LinearAddress` is `Copy`, I'm assuming that `TrieHash` is also `Copy`.